Announce deprecation of `.flutter-plugins`. (#11303)
flutter/website: src/content/release/breaking-changes/flutter-plugins-configuration.md

---

## Summary

The `flutter` tool will no longer output the legacy `.flutter-plugins` metadata
file, and only output `.flutter-plugins-dependencies`. Tools and build scripts,
such as Gradle configurations (for Android apps) that rely on the presence of
`.flutter-plugins` will need to be updated.

## Background

[In 2019][PR 45379] `.flutter-plugins-dependencies` was added as a newer file
format that replaces `.flutter-plugins`.

So a file that looks something like this:

```txt
# This is .flutter-plugins
camera=/path/to/camera/plugin
shared_preferences=shared_preferences
```

... was replaced by something like this:

```json
{
  "dependencyGraph": {
    "camera": {
      "name": "camera",
      "version": "0.10.0",
      "dependencies": {
        "flutter": "0.0.0"
      }
    },
    "shared_preferences": {
      "name": "shared_preferences",
      "version": "2.0.15",
      "dependencies": {
        "flutter": "0.0.0"
      }
    }
  },
  "flutter": {
    "frameworkRevision": "3a0f99d4f2",
    "channel": "stable"
  }
}
```

Having both files output is a source of technical debt that complicates adding
and supporting the `flutter` CLI.

## Migration guide

> Most end users of Flutter do not parse or use this file, but build artifacts

Reviewer: ```suggestion
Most Flutter developers don't parse or use this file, but build artifacts
```

> such as `settings.gradle` generated by older invocations of
> `flutter create --platforms android` may still reference `.flutter-plugins` and
> must be updated to a newer build script.

Reviewer: ```suggestion
(such as `settings.gradle`), as generated by older invocations of
`flutter create --platforms android`. These legacy files might still reference `.flutter-plugins` and
must be updated to a newer build script.
```

---
title: .flutter-plugins-dependencies replaces .flutter-plugins.
description: >-
  The deprecated `.flutter-plugins` tool file output has been replaced by
  `.flutter-plugins-dependencies`, and any build scripts or references to it
  must also be updated.
---

## Summary

The `flutter` tool will no longer output the legacy `.flutter-plugins` metadata
file, and only output `.flutter-plugins-dependencies`. Tools and build scripts,
such as Gradle configurations (for Android apps) that rely on the presence of
`.flutter-plugins` will need to be updated.

## Background

[In 2019][PR 45379] `.flutter-plugins-dependencies` was added as a newer file
format that replaces `.flutter-plugins`.

> So a file that looks something like this:

Reviewer: ```suggestion
So a file that looked something like this:
```

```txt
# This is .flutter-plugins
camera=/path/to/camera/plugin
shared_preferences=shared_preferences
```

... was replaced by something like this:

```json
{
  "dependencyGraph": {
    "camera": {
      "name": "camera",
      "version": "0.10.0",
      "dependencies": {
        "flutter": "0.0.0"
      }
    },
    "shared_preferences": {
      "name": "shared_preferences",
      "version": "2.0.15",
      "dependencies": {
        "flutter": "0.0.0"
      }
    }
  },
  "flutter": {
    "frameworkRevision": "3a0f99d4f2",
    "channel": "stable"
  }
}
```

> Having both files output is a source of technical debt that complicates adding

Reviewer 2: adding what? adding to?

Author: Oops, done!
